jplag/src/main/java/de/jplag/SubmissionSetBuilder.java

```diff
@@ -92,13 +109,75 @@ private void verifyRootdirExistence(File rootDir) throws ExitException {
     }
 
     /**
-     * Verify the basecode name.
+     * Try to load the basecode under the assumption of being a path.
+     * @return Base code submission if the option value can be interpreted as global path, else {@code null}.
+     * @throws ExitException when the option value is a path with errors.
+     */
+    private Submission tryLoadBaseCodeAsPath() throws ExitException {
+        String baseCodeName = options.getBaseCodeSubmissionName().get();
+        File basecodeSubmission = new File(baseCodeName);
+        if (!basecodeSubmission.exists()) {
+            return null;
+        }
+
+        String errorMessage = isExcludedEntry(basecodeSubmission);
+        if (errorMessage != null) {
+            throw new BasecodeException(errorMessage); // Stating an excluded path as basecode isn't very useful.
+        }
+
+        try {
+            return processDirEntry(basecodeSubmission);
+
+        } catch (SubmissionException ex) {
+            throw new BasecodeException(ex.getMessage()); // Change thrown exception to basecode exception.
+
+        } catch (ExitException ex) {
+            throw ex;
+        }
+    }
+
+    /**
+     * Try to load the basecode under the assumption of being a sub-directory in the user submissions directory.
+     * @return Base code submission if the option value can be interpreted as a sub-directory, else {@code null}.
+     * @throws ExitException when the option value is a sub-directory with errors.
      */
-    private void verifyBaseCodeName() throws ExitException {
-        String name = options.getBaseCodeSubmissionName();
+    private Submission tryLoadBaseCodeAsRootSubDirectory(Map<String, Submission> foundSubmissions) throws ExitException {
+        String baseCodeName = options.getBaseCodeSubmissionName().get();
+
+        // Is the option value a single name after trimming spurious separators?
+        String name = trimLeft(trimRight(baseCodeName, File.separatorChar), File.separatorChar);
+        if (name.contains(File.separator)) return null; // Not a single name, cannot be a sub-directory.
+
         if (name.contains(".")) {
             throw new BasecodeException("The basecode directory name \"" + name + "\" cannot contain dots!");
         }
+
+        // Grab the basecode submission from the regular submissions.
+        return foundSubmissions.get(name);
+    }
+
+    /**
+     * Trim characters k from the left of the string.
+     */
+    private String trimLeft(String s, char k) {
+        int i = 0;
+        while (i < s.length() && s.charAt(i) == k) {
+            i++;
+        }
+        if (i == 0) return s;
+        return s.substring(i);
+    }
+
+    /**
+     * Trim characters k from the right of the string.
+     */
+    private String trimRight(String s, char k) {
+        int i = s.length() - 1;
+        while (i >= 0 && s.charAt(i) == k) {
+            i--;
+        }
+        if (i == s.length() - 1) return s;
+        return s.substring(0, i + 1);
```

Comment:
Three issues here:
- I prefer speaking variable names (sorry for being pedantic)
- These methods are overly generic but only ever used to remove the file separator
- They can be merged and simplified
Thus I propose replacing them with:
```java
private String trimFileSeparators(String text) {
    String trimmedText = text;
    while (trimmedText.startsWith(File.separator)) {
        trimmedText = trimmedText.substring(1);
    }
    while (trimmedText.endsWith(File.separator)) {
        trimmedText = trimmedText.substring(0, trimmedText.length() - 1);
    }
    return trimmedText;
}
````

Reply:
Yeah, I assumed Java would have a basic trim functions but I was wrong. Then I got stuck on that road implementing them myself.

Fixed by inlining your suggestion in the calling function.